contrib/epee/include/net/levin_protocol_handler_async.h

```diff
@@ -595,9 +626,15 @@ class async_protocol_handler
                             << ", ver=" << head.m_protocol_version);
 
     uint64_t ticks_start = misc_utils::get_tick_count();
+    size_t prev_size = 0;
 
     while(!boost::interprocess::ipcdetail::atomic_read32(&m_invoke_buf_ready) && !m_deletion_initiated && !m_protocol_released)
     {
+      if(m_cache_in_buffer.size() - prev_size >= MIN_BYTES_WANTED)
+      {
+        prev_size = m_cache_in_buffer.size();
+        ticks_start = misc_utils::get_tick_count();
+      }
```

Comment:
Yeah, this is the synchronous handler, which Monero doesn't use. And the call_run_once_service_io does all the I/O inline in this same thread.